### PR TITLE
Cybernetic/Upgraded Cybernetic Ear replacement

### PR DESCRIPTION
**Pull request by RandolfTheMeh**
## About The Pull Request

I was contacted by an individual, Detective Google, who wanted to make this PR but was gitbanned, so he sent me the files. They all seemed in order, and from what I understand he was in communication with Cobby in regards to this PR, and giving it a look myself it all seems in order. Some minor changes were made, but that all aside, about the PR.

Cybernetic ears are now added to the medical protolathe and to the robotics fabricator, which offer a minor boost in resisting ear damage, and small decreases to the time spent deaf from such affects. Upgraded cybernetic ears take halved ear damage, and are deaf for only half the time, but require silver to make. Neither organ decays, as per cybernetic organs, but take a considerable amount of damage when EMPed.

## Why It's Good For The Game

Ears have long deserved some attention in regards to augmentation, and although inacusiate exists to fix any and all damage, an augmentation that was more resilient to that damage in the first place seems apt.

## Changelog
:cl:
add: Cybernetic and Upgraded Cybernetic ears, which are hardier versions of normal ears. Regular ones reduce deafness/damage by 10%, and upgraded ones reduce deafness/damage by 50% but cost more resources.
/:cl:

### CODE REVIEW

**Comment by carlarctg** [comment]
i think this just leaves brains and stomachs without an augmented version

**Comment by RandolfTheMeh** [comment]
There's the question of whether inacusiate should work on cybernetic organs. I'll leave it as is, currently, but I'm leaning towards preventing inacusiate from working on them.

**Comment by RandolfTheMeh** [comment]
Oh right and the sprites
aaaaswerf

File: code/modules/research/techweb/all_nodes.dm [hunk 1]
```diff
@@ -481,7 +481,7 @@
 	display_name = "Cybernetic Organs"
 	description = "We have the technology to rebuild him."
 	prereq_ids = list("adv_biotech")
-	design_ids = list("cybernetic_heart", "cybernetic_liver", "cybernetic_lungs")
+	design_ids = list("cybernetic_ears", "cybernetic_heart", "cybernetic_liver", "cybernetic_lungs", )
```
Comment (Dennok) [inline code/modules/research/techweb/all_nodes.dm]: superfluous comma?

Comment (nemvar) [reply code/modules/research/techweb/all_nodes.dm]: This comma might actually cause issues. If I remember correctly, this adds a null to the list.

File: code/modules/surgery/organs/ears.dm [hunk 2]
```diff
@@ -133,3 +133,27 @@
...
+	desc = "an advanced cybernetic ear, surpassing the performance of organic ears"
+	damage_multiplier = 0.5
+	bang_protect = 1
+
+/obj/item/organ/ears/cybernetic/emp_act(severity)
+	. = ..()
+	if(. & EMP_PROTECT_SELF)
+		return
+	switch(severity)
+		if(1)
```
Comment (nemvar) [inline code/modules/surgery/organs/ears.dm]: Defines please.

**Comment by LoGIRz** [comment]
Question: if a universal translator exists as a personal AI software option shouldn't cybernetic ears allow you to understand any language as well?

**Comment by CRITAWAKETS** [comment]
> Question: if a universal translator exists as a personal AI software option shouldn't cybernetic ears allow you to understand any language as well?

It was planned to be the effect of the upgraded ears, but speaking and understanding languages arent seperate things.

**Comment by knacker48** [comment]
If you do end up making inacusiate not work on cyber ear will you at least give an alternative way to fix them? Cause it would suck if you just went through the surgery and got emped and could do nothing to fix it

**Comment by LoGIRz** [comment]
Speaking any known language is separately handled by #45747 from a vending machine you have to order from cargo and hack. I presume there is a separate option to understand any heard language but not speak it

File: code/modules/surgery/organs/ears.dm (hunk 2 above)
Comment (RandolfTheMeh) [reply code/modules/surgery/organs/ears.dm]: This is ripped straight from liver cybernetics code. Technically doesn't even have to exist, I'm not sure why they made a switch case for two cases.

**Comment by optimumtact** [comment]
power creep

**Comment by TerraGS** [comment]
Needs a cat ear variant for turning people into cybernetic cat girls.

**Comment by lifeyouristhis** [comment]
Why the cybernetic / upgraded cybernetic, though (questioning the current system, not the PR). They are separated from themselves by just half a minute of research points. It's pretty pointless.

**Comment by ATH1909** [comment]
> Speaking any known language is separately handled by #45747 from a vending machine you have to order from cargo and hack.

coaix, the original creator of languages, has said that robotic voiceboxes currently shouldn't give you knowledge of all languages (and if they do, it's a bug). Do you have any evidence of this?

**Comment by NoxVS** [comment]
Just had a random idea. Would it be possible to make it so you can hear whispers as if it was spoken normally if you have advanced cybernetic ears? So someone whispers on the other end of your screen and you can still hear them

**Comment by Dennok** [comment]
Nice idea! But this must be opposite to bang resist.

**Comment by ExcessiveUseOfCobblestone** [comment]
no flashbang protection.

**Comment by GuyonBroadway** [comment]
Giving it a cat variant would mean felinids have a method of utterly negating thier intrinsic weakness to flashbangs and sonic powder. 

Plus felinids are a disabled and depricated feature anyways.

**Comment by Dennok** [comment]
> Giving it a cat variant would mean felinids have a method of utterly negating thier intrinsic weakness to flashbangs and sonic powder.
> 
> Plus felinids are a disabled and depricated feature anyways.

You already can remove ears, and this make you immune to loud sounds.
If no, something is bad.

**Comment by GuyonBroadway** [comment]
You can but it comes with the downside of being deaf and your average felinid player would probably get all unnecessary about losing a few pixels off the top of thier head.

**Comment by ExcessiveUseOfCobblestone** [comment]
CL update

File: code/modules/surgery/organs/ears.dm [hunk 3]
```diff
@@ -133,3 +133,22 @@
...
+	icon_state = "ears-c"
+	desc = "a basic cybernetic designed to mimic the operation of ears."
+	damage_multiplier = 0.9
+	organ_flags = ORGAN_SYNTHETIC
+
+/obj/item/organ/ears/cybernetic/upgraded
+	name = "upgraded cybernetic ears"
+	icon_state = "ears-c-u"
+	desc = "an advanced cybernetic ear, surpassing the performance of organic ears"
+	damage_multiplier = 0.5
```
Comment (coiax) [inline code/modules/surgery/organs/ears.dm]: You said that the upgraded ears were bang proof, but I don't see it in these stats.

Comment (RandolfTheMeh) [reply code/modules/surgery/organs/ears.dm]: Yeah, my bad. Didn't redo the body of the PR, since bang-proof ears was nixed.

**Comment by RandolfTheMeh** [comment]
I redownloaded the surgery dmi and it's still given me an icon diff, not like I can tell if it was actually changed since the alpha of only a few pixels were changed on them, so I'm not sure if it's actually been updated, but I took it from the github repo after being notified of the conflict.

**Comment by SpaceManiac** [comment]
Merged for you, `git pull` to update your local.

**Comment by RandolfTheMeh** [comment]
> 
> 
> Merged for you, `git pull` to update your local.
> 
> https://github.com/tgstation/tgstation/wiki/Resolving-icon-conflicts for how to do this yourself

Can't thank you enough for both the assistance and the resources to help avoid this in the future!

File: code/modules/surgery/organs/ears.dm [hunk 4]
```diff
@@ -133,3 +133,22 @@
...
+
+/obj/item/organ/ears/cybernetic/emp_act(severity)
+	. = ..()
+	if(. & EMP_PROTECT_SELF)
+		return
+	damage += 40/severity
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/surgery/organs/ears.dm]: am I dumb or does higher severity do LOWER damage.

Prob should just have this as  `(20 * severity)`?

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
once above addressed R2M

File: code/modules/surgery/organs/ears.dm (hunk 4 above)
Comment (RandolfTheMeh) [reply code/modules/surgery/organs/ears.dm]: Higher severity should do lower damage. Severity one is severe, severity two is mild. I didn't make the convention for this.

**Pull request merged**